File: Data Structures & Algorithms/top-k-elements-in-list/submission-0.ts
class Solution {
    /**
     * @param {number[]} nums
     * @param {number} k
     * @return {number[]}
     */
    topKFrequent(nums: number[], k: number): number[] {
        const numsMap = new Map<number, number>()
        for (var num of nums) {
            numsMap.set(num, (numsMap.get(num) ?? 0) + 1)
        }

        const result = [...numsMap.entries()].sort(([fk, fv], [sk, sv]) => sv - fv).filter((v, i) => {            
            if (i >= k) return false
            return true
        }).map((k, v) => k[0])

        return result
    }
}

File: Data Structures & Algorithms/products-of-array-discluding-self/submission-0.cs
public class Solution {
    public int[] ProductExceptSelf(int[] nums) {
        int[] returnArr = new int[nums.Length];

        for (int i = 0; i < nums.Length; i++)
        {
            bool set = false;
            for (int b = 0; b < nums.Length; b++)
            {
                if (i == b)
                {
                    continue;
                }

                if (!set)
                {
                    returnArr[i] = nums[b];
                    set = true;
                    continue;
                }

                returnArr[i] *= nums[b];
            }
        }

        return returnArr;
    }
}
